app/bundles/EmailBundle/MonitoredEmail/Search/ContactFinder.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+/*
+ * @copyright   2017 Mautic Contributors. All rights reserved
+ * @author      Mautic, Inc.
+ *
+ * @link        https://mautic.org
+ *
+ * @license     GNU/GPLv3 http://www.gnu.org/licenses/gpl-3.0.html
+ */
+
+namespace Mautic\EmailBundle\MonitoredEmail\Search;
+
+use Mautic\EmailBundle\Entity\Stat;
+use Mautic\EmailBundle\Entity\StatRepository;
+use Mautic\EmailBundle\MonitoredEmail\Processor\Address;
+use Mautic\LeadBundle\Entity\LeadRepository;
+use Psr\Log\LoggerInterface;
+
+class ContactFinder
+{
+    /**
+     * @var StatRepository
+     */
+    protected $statRepository;
+
+    /**
+     * @var LeadRepository
+     */
+    protected $leadRepository;
+
+    /**
+     * @var LoggerInterface
+     */
+    protected $logger;
+
+    /**
+     * ContactFinder constructor.
+     *
+     * @param StatRepository  $statRepository
+     * @param LeadRepository  $leadRepository
+     * @param LoggerInterface $logger
+     */
+    public function __construct(StatRepository $statRepository, LeadRepository $leadRepository, LoggerInterface $logger)
+    {
+        $this->statRepository = $statRepository;
+        $this->leadRepository = $leadRepository;
+        $this->logger         = $logger;
+    }
+
+    /**
+     * @param      $returnPathEmail
+     * @param null $contactEmail
+     *
+     * @return Result
+     */
+    public function find($contactEmail, $returnPathEmail = null)
+    {
+        $this->logger->debug("MONITORED EMAIL: Searching for a contact $contactEmail/$returnPathEmail");
+
+        // We have a return path email so let's figure out who it originated to
+        if ($returnPathEmail && $hash = Address::parseAddressForStatHash($returnPathEmail)) {
+            if ($result = $this->findByHash($hash)) {
```

Comment:
`findByHash` always returns a `Result`, so there is a bug (should not return an empty `Result`) or this condition is useless.